frontend/src/app/components/faucet/faucet.component.ts

```diff
@@ -1,66 +1,89 @@
-import { Component, OnDestroy, OnInit } from "@angular/core";
-import { FormBuilder, FormGroup, ValidationErrors, Validators } from "@angular/forms";
-import { StorageService } from '../../services/storage.service';
-import { ServicesApiServices } from '../../services/services-api.service';
-import { AudioService } from '../../services/audio.service';
-import { StateService } from '../../services/state.service';
-import { Subscription, tap } from "rxjs";
-import { HttpErrorResponse } from "@angular/common/http";
+import { Component, OnDestroy, OnInit, ChangeDetectorRef } from "@angular/core";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { Subscription } from "rxjs";
+import { StorageService } from "../../services/storage.service";
+import { ServicesApiServices } from "../../services/services-api.service";
 import { getRegex } from "../../shared/regex.utils";
+import { StateService } from "../../services/state.service";
 import { WebsocketService } from "../../services/websocket.service";
+import { AudioService } from "../../services/audio.service";
+import { HttpErrorResponse } from "@angular/common/http";
 
 @Component({
   selector: 'app-faucet',
   templateUrl: './faucet.component.html',
   styleUrls: ['./faucet.component.scss']
 })
 export class FaucetComponent implements OnInit, OnDestroy {
-  user: any;
-  loading: boolean = true;
+  loading = true;
+  error: string = '';
+  user: any = undefined;
+  txid: string = '';
+ 
+  faucetStatusSubscription: Subscription;
   status: {
-    address?: string,
-    access: boolean
-    min: number,
-    user_max: number,
-    user_requests: number,
+    min: number; // minimum amount to request at once (in sats)
+    max: number; // maximum amount to request at once
+    address?: string; // faucet address
   } | null = null;
-  error = '';
   faucetForm: FormGroup;
-  txid = '';
 
   mempoolPositionSubscription: Subscription;
   confirmationSubscription: Subscription;
 
   constructor(
-    private stateService: StateService,
+    private cd: ChangeDetectorRef,
     private storageService: StorageService,
     private servicesApiService: ServicesApiServices,
-    private websocketService: WebsocketService,
-    private audioService: AudioService,
     private formBuilder: FormBuilder,
+    private stateService: StateService,
+    private websocketService: WebsocketService,
+    private audioService: AudioService
   ) {
+    this.faucetForm = this.formBuilder.group({
+      'address': ['', [Validators.required, Validators.pattern(getRegex('address', 'testnet4'))]],
+      'satoshis': [0, [Validators.required, Validators.min(0), Validators.max(0)]]
+    });
+  }
+
+  ngOnDestroy() {
+    if (this.faucetStatusSubscription) {
+      this.faucetStatusSubscription.unsubscribe();
```

Comment:
there's no need to unsubscribe from HttpClient observables (it happens automatically)